build-logic/jvm/src/main/kotlin/gradlebuild.unittest-and-compile.gradle.kts

```diff
@@ -198,10 +198,6 @@ fun configureTests() {
         }
     }
 
-    if (project.testDistributionEnabled()) {
```

Comment:
Since it's no longer a project-plugin, we no longer gave to apply it on individual sub projects but just have to load it once as a settings plugin.

This change will also get rid of the `ClassCastException`s which could rarely be observed in GBT.